modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -2210,6 +2228,45 @@ bool CvVideoWriter_FFMPEG::writeFrame( const unsigned char* data, int step, int
     return ret;
 }
 
+bool CvVideoWriter_FFMPEG::writeHWFrame(cv::InputArray input) {
+#if USE_AV_HW_CODECS
+    if (!video_st->codec->hw_frames_ctx)
+        return false;
+
+    // check that current OpenCL context initilized on same media device as codec
+    //if (!hw_check_opencl_context(video_st->codec->hw_device_ctx))
+    //    return false;
```

Comment:
Perhaps robust check should check UMat's OpenCL context.

Reply:
It would require code like below?
```
        CV_Assert(u.u && u.u->currAllocator == ocl::getOpenCLAllocator());
        CV_Assert(std::static_pointer_cast<ocl::Context>(u.u->allocatorContext)->ptr() == ocl_context.getContext().ptr());
```
I believe SetKernelArgument will throw exception if wrong context, and this code under try/catch - is it ok or better check context?

Reply:
Not sure that OpenCL CTS has checks for "wrong context" cases.

I would prefer to raise OpenCV exception instead of crash in OpenCL runtime internals.